project/src/main/java/com/google/sps/utilities/InstanceDatabaseWrapper.java

```diff
@@ -0,0 +1,84 @@
+package com.google.sps.utilities;
+
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.sps.data.User;
+import java.util.ArrayList;
+import java.util.List;
+
+public class InstanceDatabaseWrapper implements DatabaseWrapper {
+  private String instanceID;
+  private String databaseID;
+  private static final String USER_TABLE = "Users";
+
+  public InstanceDatabaseWrapper(String instanceID, String databaseID) {
+    this.instanceID = instanceID;
+    this.databaseID = databaseID;
+  }
+
+  /**
+   * Given a user, insert a row with all available fields into the DB
+   *
+   * @param user the user to be inserted; user's ID field should not exist in DB
+   */
+  public void insertUser(User user) {
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceID, databaseID);
+    DatabaseClient dbClient = spanner.getDatabaseClient(db);
```

Comment:
Can this be a class variable and initialize in constructor? So the all the actions can just call dbClient.

Reply:
Ack - I will try to make a new PR from master since this code is already on master.

Reply:
The question @dash102 and I have about this is how to call spanner.close() after all operations are complete within the DatabaseWrapper class if we move this to the constructor.
A possible way is to create helper function closeSpanner() in DatabaseWrapper and force all servlets to call this method at the end but this does not seem clean.